### PR TITLE
AMI430: fix coil_constant-related bugs of current- and field- parameters

### PR DESCRIPTION
**Pull request by astafan8**
Fixes:
* `has_current_rating` argument/attribute should only matter for `field_rating` parameter, not for `field_ramp_limit` parameter.
* Update `scale` of `field_limit` parameter when a `coil_constant` is set
* Add tests for field* and current* parameters so that they are proportional via coil_constant
* Add tests reducing `ramp_rate` if new `current_ramp_limit` or `field_ramp_limit` is lower than `ramp_rate`
* Apply PEP8
* Fix AMI430 simulation yaml file

This PR might end up resolving https://github.com/QCoDeS/Qcodes/issues/1525.

### CODE REVIEW

**Comment by astafan8** [comment]
@damazter Could you see if this PR fixes the coil-constant issue https://github.com/QCoDeS/Qcodes/issues/1525 ?

**Comment by jenshnielsen** [comment]
Did we ever figure out why this difference exists. Afaik it had something todo with the driver/docs version on AMIs webpage being out of date and the new versions only being on bitbucket? It would be good to at least update the comments about this.

**Comment by astafan8** [comment]
@jenshnielsen with the "difference" you mean the `has_current_rating` attribute? and with "update comment" you mean to update the following one?:
https://github.com/QCoDeS/Qcodes/blob/4c4f8f5edd9cd94fabf3ceb07cb4df556d347ece/qcodes/instrument_drivers/american_magnetics/AMI430.py#L177-180

**Comment by WilliamHPNielsen** [comment]
That comment still reflects my knowledge, unfortunately. But I have not compared firmware versions of different models, so there is still the possibility that we'll be able to figure out roughly in what firmware version that feature was added/removed.

**Comment by astafan8** [comment]
Thanks to @damazter @damazter2 rigorous testing of this change, we now confirm that the coil-constant related bugs are solved. By request of Damaz, two tests were added: one that runs the coil-constant/field/current-limit tests with arbitrary values and a number of times, and one new test for the fact that ramp_rate is reduced when ramp_limit is made lower than the current (now) ramp_rate.

After CI shows green, I am going to merge this. @QCoDeS/core any objections?

File: qcodes/tests/drivers/test_ami430.py [hunk 1]
```diff
@@ -440,6 +555,230 @@ def test_blocking_ramp_parameter(current_driver, caplog):
...
+    Note that the 'factor's are randomized "manually" because of the
+    possibility to hit the limits of the parameters.
+    """
+    with warnings.catch_warnings():
+        # this is to avoid AMI430Warning about "maximum ramp rate", which
+        # may show up but is not relevant to this test
+        warnings.simplefilter('ignore', category=AMI430Warning)
+
+        test_current_and_field_params_interlink_at_init(ami430)
+
```
Comment (jenshnielsen) [inline qcodes/tests/drivers/test_ami430.py]: Would it make sense to make this a bit more DRY by looping over 3 tuples or pytest parametrize ?

Comment (astafan8) [reply qcodes/tests/drivers/test_ami430.py]: sure. i can probably hard-code some values and use permutations from itertools or smth.

but only if you insist :)

Comment (astafan8) [reply qcodes/tests/drivers/test_ami430.py]: @jenshnielsen ? :)

Comment (jenshnielsen) [reply qcodes/tests/drivers/test_ami430.py]: I would prefer but I don't insist. That's why its a question :)

Comment (astafan8) [reply qcodes/tests/drivers/test_ami430.py]: ok :)

**Pull request merged**